### PR TITLE
[no_severity] Fix the query to exclude bugs that recently changed components

### LINKED ISSUE #1250: Don't autonag claiming bugs "haven't had a severity set for 4 weeks" if they have moved component/product in that timeframe
People shouldn't be held responsible for the triage of bugs that they had no opportunity to see, given that until recently they weren't in their bailiwick.

### PR DESCRIPTION
**Pull request by suhaibmujahid**
Fixes #1250

The query:
- [Before this PR](https://bugzilla.mozilla.org/buglist.cgi?include_fields=triage_owner&include_fields=flags&include_fields=comments.creator&include_fields=comments.creation_time&include_fields=id&include_fields=summary&include_fields=groups&include_fields=product&include_fields=component&keywords=intermittent-failure&keywords_type=nowords&email2=wptsync%40mozilla.bugs&emailreporter2=1&emailtype2=notequals&resolution=---&f31=bug_type&o31=equals&v31=defect&f32=flagtypes.name&o32=notsubstring&v32=needinfo%3F&f33=bug_severity&o33=anyexact&v33=--%2C+n%2Fa&j2=OR&f2=OP&j3=AND&f3=OP&f4=product&o4=changedbefore&v4=-28d&n5=1&f5=product&o5=changedafter&v5=-28d&f6=CP&j7=AND&f7=OP&f8=component&o8=changedbefore&v8=-28d&n9=1&f9=component&o9=changedafter&v9=-28d&f10=CP&j11=AND&f11=OP&f12=creation_ts&o12=lessthaneq&v12=-28d&n13=1&f13=product&o13=everchanged&n14=1&f14=component&o14=everchanged&f15=CP&f16=CP&n20=1&j20=OR&f20=OP&f21=bug_severity&o21=changedfrom&v21=critical&f22=bug_severity&o22=changedfrom&v22=major&f23=bug_severity&o23=changedfrom&v23=blocker&f30=CP&f34=creation_ts&o34=greaterthan&v34=-24m&f35=status_whiteboard&o35=notsubstring&v35=%5Bno-nag%5D&f36=keywords&o36=nowords&v36=meta&product=Core&product=Developer+Infrastructure&product=DevTools&product=External+Software+Affecting+Firefox&product=Fenix&product=Firefox&product=Firefox+for+iOS&product=Firefox+Build+System&product=GeckoView&product=JSS&product=NSPR&product=NSS&product=Remote+Protocol&product=Testing&product=Toolkit&product=Web+Compatibility&product=WebExtensions)
- [After this PR](https://bugzilla.mozilla.org/buglist.cgi?include_fields=triage_owner&include_fields=flags&include_fields=comments.creator&include_fields=comments.creation_time&include_fields=id&include_fields=summary&include_fields=groups&include_fields=product&include_fields=component&keywords=intermittent-failure&keywords_type=nowords&email2=wptsync%40mozilla.bugs&emailreporter2=1&emailtype2=notequals&resolution=---&f31=bug_type&o31=equals&v31=defect&f32=flagtypes.name&o32=notsubstring&v32=needinfo%3F&f33=bug_severity&o33=anyexact&v33=--%2C+n%2Fa&j2=OR&f2=OP&j3=AND&f3=OP&f4=product&o4=changedbefore&v4=-28d&n5=1&f5=product&o5=changedafter&v5=-28d&n6=1&f6=component&o6=changedafter&v6=-28d&f7=CP&j8=AND&f8=OP&f9=component&o9=changedbefore&n10=1&f10=product&o10=changedafter&v10=-28d&v9=-28d&n11=1&f11=component&o11=changedafter&v11=-28d&f12=CP&j13=AND&f13=OP&f14=creation_ts&o14=lessthaneq&v14=-28d&n15=1&f15=product&o15=everchanged&n16=1&f16=component&o16=everchanged&f17=CP&f18=CP&n20=1&j20=OR&f20=OP&f21=bug_severity&o21=changedfrom&v21=critical&f22=bug_severity&o22=changedfrom&v22=major&f23=bug_severity&o23=changedfrom&v23=blocker&f30=CP&f34=creation_ts&o34=greaterthan&v34=-24m&f35=status_whiteboard&o35=notsubstring&v35=%5Bno-nag%5D&f36=keywords&o36=nowords&v36=meta&product=Core&product=Developer+Infrastructure&product=DevTools&product=External+Software+Affecting+Firefox&product=Fenix&product=Firefox&product=Firefox+for+iOS&product=Firefox+Build+System&product=GeckoView&product=JSS&product=NSPR&product=NSS&product=Remote+Protocol&product=Testing&product=Toolkit&product=Web+Compatibility&product=WebExtensions)

The query is very long. I think it would be clearer to do the "component change" filtering on the client side. It would be ideal to do this refactoring as part of #1551. @marco-c wdyt?

## Checklist

- [ ] Type annotations added to new functions
- [ ] Docs added to functions touched in main classes
- [ ] Dry-run produced the expected results
- [ ] The [`to-be-announced`](https://github.com/mozilla/relman-auto-nag/labels/to-be-announced) tag added if this is worth announcing

### CODE REVIEW

File: auto_nag/scripts/workflow/no_severity.py [hunk 1]
```diff
@@ -226,30 +226,38 @@ def get_bz_params(self, date):
...
+                    "j8": "AND",
+                    "f8": "OP",
                     "f9": "component",
-                    "o9": "changedafter",
+                    "o9": "changedbefore",
+                    "n10": 1,
+                    "f10": "product",
+                    "o10": "changedafter",
+                    "v10": second,
                     "v9": second,
```
Comment (marco-c) [inline auto_nag/scripts/workflow/no_severity.py:242]: could you move this line to after the "o9" line so the "9" are close together?

**Comment by marco-c** [comment]
> The query is very long. I think it would be clearer to do the "component change" filtering on the client side. It would be ideal to do this refactoring as part of https://github.com/mozilla/relman-auto-nag/pull/1551. @marco-c wdyt?

Yes, this is pretty hard to review!

File: auto_nag/scripts/workflow/no_severity.py (hunk 1 above)
Comment (suhaibmujahid) [reply auto_nag/scripts/workflow/no_severity.py]: Done in a56698a3c414f7eb2ba0b2b6f739ad8057fec302. It was by mistake.

**Comment by suhaibmujahid** [comment]
> Yes, this is pretty hard to review!

Changing the numbers makes the diff messy. Below is a visualization to diff:


## Before
Screen Shot 2022-10-28 at 7 39 33 PM

## After
Screen Shot 2022-10-28 at 7 39 15 PM

**Review by marco-c — APPROVED** [review]
Thanks, the image makes it much easier.

**Pull request merged**